gid.c

```diff
@@ -0,0 +1,22 @@
+#include <sys/types.h>
+#include <grp.h>
+#include "uidgid.h"
+#include "subfd.h"
+#include "substdio.h"
+#include "exit.h"
+
+int
+initgid(group)
+  char *group;
+{
+  struct group *gr;
+  gr = getgrnam(group);
+  if (!gr) {
+    substdio_puts(subfderr,"fatal: unable to find group ");
+    substdio_puts(subfderr,group);
+    substdio_puts(subfderr,"\n");
+    substdio_flush(subfderr);
+    _exit(111);
```

Comment:
When one of the qmail programs starts, fails a lookup, and exits 111, what does that look like? Do we see log entries? As admins, how will we know what happened and what to do about it?

At a glance, looks like the programs to be concerned with are:

- `qmail-lspawn`
- `qmail-queue`
- `qmail-rspawn`
- `qmail-start`

Reply:
If you run `qmail-start` with a missing user, the program produces the following output:

    qmail-start
    fatal: unable to find user qmaill

The other programs or missing users produce the same result.  Note in the specific case of `qmail-start`, this is an improvement over the prior behavior, where if you built on a system with the qmail system accounts but then ran it on a system where they were not present, it would silently fail.